test/entrypoint.test.ts

```diff
@@ -543,6 +543,47 @@ describe('EntryPoint', function () {
         expect(await getBalance(account.address)).to.eq(inititalAccountBalance)
       })
 
+      it('account should pay a penalty for requiring too much gas and leaving it unused', async function () {
+        if (process.env.COVERAGE != null) {
+          return
+        }
+        const iterations = 10
+        const count = await counter.populateTransaction.gasWaster(iterations, '')
```

Comment:
I think that a good estimate to the callGasUsed is 
`estimateGas({from:entryPoint, to: account.address, data: op1.callData})`